Mathlib/Geometry/Manifold/VectorField.lean

```diff
@@ -0,0 +1,317 @@
+  {I' : ModelWithCorners 𝕜 E' H'}
+  {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M']
+  {H'' : Type*} [TopologicalSpace H''] {E'' : Type*} [NormedAddCommGroup E''] [NormedSpace 𝕜 E'']
+  {I'' : ModelWithCorners 𝕜 E'' H''}
+  {M'' : Type*} [TopologicalSpace M''] [ChartedSpace H'' M'']
+  {f : M → M'} {s t : Set M} {x x₀ : M}
+
+instance {n : ℕ} [n.AtLeastTwo] [IsManifold I (minSmoothness 𝕜 (ofNat(n))) M] :
+    IsManifold I (ofNat(n)) M :=
+  IsManifold.of_le (n := minSmoothness 𝕜 n) le_minSmoothness
+
+instance [IsManifold I (minSmoothness 𝕜 1) M] :
+    IsManifold I 1 M :=
+  IsManifold.of_le (n := minSmoothness 𝕜 1) le_minSmoothness
+
+instance [IsManifold I (minSmoothness 𝕜 3) M] :
+    IsManifold I (minSmoothness 𝕜 2) M :=
+  IsManifold.of_le (n := minSmoothness 𝕜 3) (minSmoothness_monotone (by norm_num))
+
+instance [IsManifold I (minSmoothness 𝕜 2) M] :
+    IsManifold I (minSmoothness 𝕜 1) M :=
+  IsManifold.of_le (n := minSmoothness 𝕜 2) (minSmoothness_monotone (by norm_num))
+
+
+namespace VectorField
+
+section
+
+/-! ### Pullback of vector fields in manifolds -/
+
+open ContinuousLinearMap
+
+variable {V W V₁ W₁ : Π (x : M'), TangentSpace I' x}
+variable {c : 𝕜} {m n : WithTop ℕ∞} {t : Set M'} {y₀ : M'}
+
+variable (I I') in
+/-- The pullback of a vector field under a map between manifolds, within a set `s`. -/
+def mpullbackWithin (f : M → M') (V : Π (x : M'), TangentSpace I' x) (s : Set M) (x : M) :
+    TangentSpace I x :=
+  (mfderivWithin I I' f s x).inverse (V (f x))
+
+variable (I I') in
+/-- The pullback of a vector field under a map between manifolds. -/
+def mpullback (f : M → M') (V : Π (x : M'), TangentSpace I' x) (x : M) :
+    TangentSpace I x :=
+  (mfderiv I I' f x).inverse (V (f x))
+
+lemma mpullbackWithin_apply :
+    mpullbackWithin I I' f V s x = (mfderivWithin I I' f s x).inverse (V (f x)) := rfl
+
+lemma mpullbackWithin_smul_apply :
+    mpullbackWithin I I' f (c • V) s x = c • mpullbackWithin I I' f V s x := by
+  simp [mpullbackWithin_apply]
+
+lemma mpullbackWithin_smul :
+    mpullbackWithin I I' f (c • V) s = c • mpullbackWithin I I' f V s := by
+  ext x
+  simp [mpullbackWithin_apply]
+
+lemma mpullbackWithin_add_apply :
+    mpullbackWithin I I' f (V + V₁) s x =
+      mpullbackWithin I I' f V s x + mpullbackWithin I I' f V₁ s x := by
+  simp [mpullbackWithin_apply]
+
+lemma mpullbackWithin_add :
+    mpullbackWithin I I' f (V + V₁) s =
+      mpullbackWithin I I' f V s + mpullbackWithin I I' f V₁ s := by
+  ext x
+  simp [mpullbackWithin_apply]
+
+lemma mpullbackWithin_neg_apply :
+    mpullbackWithin I I' f (-V) s x = - mpullbackWithin I I' f V s x := by
+  simp [mpullbackWithin_apply]
+
+lemma mpullbackWithin_neg :
+    mpullbackWithin I I' f (-V) s = - mpullbackWithin I I' f V s := by
+  ext x
+  simp [mpullbackWithin_apply]
+
+lemma mpullback_apply :
+    mpullback I I' f V x = (mfderiv I I' f x).inverse (V (f x)) := rfl
+
```

Comment:
How about adding the `smul` versions also? (You only have them for `mpullbackWithin`.)